components/VideoGrid.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div class="video-grid__wrapper">
+    <v-text-field
+      class="video-grid__search"
+      :label="`Search in all ${videos.length} videos`"
+      v-model="search"
+      @input="handleQueryChange"
+      single-line
+      hide-details
+      prepend-icon="mdi-magnify"
+    />
+    <div class="video-grid__list">
+      <VideoGridItem
+        v-for="video in paginatedVideos"
+        :key="video.id"
+        :video="video"
+        :highlighted-term="search"
+        @click.native="$router.push(`/watch/${video.id}`)"
+      />
+    </div>
+    <div class="text-center mt-5">
+      <v-pagination
+        :key="search"
+        color="green"
+        v-model="page"
+        :length="pagesCount"
+        :total-visible="7"
+      />
```

Comment:
Where is the pagination component ?

Reply:
Vuetify https://v2.vuetifyjs.com/en/components/paginations/